Create setup_and_pull_general.py

Abdelaziz-Serour opened the pull request:
Replaced separate Windows (.bat) and Linux (.sh) model-setup scripts with a single cross-platform Python script: setup_and_pull_general.py
The new script:
1. Loads env vars from .env (quote-safe, comment-aware).
2. Sets OLLAMA_HOST from MODEL_BASE_URL.
3. Validates required vars: MODEL_NAME, CODING_AGENT_MODEL_NAME, EMBEDDING_MODEL.
4. Verifies ollama CLI presence.
5. Pulls all required models (ollama pull ...) with clear logging and failure modes.
6. Supports custom env path via --env-file.

Abdelaziz-Serour committed:
Create setup_and_pull_general.py

omar-abdelgawad commented on setup_and_pull_general.py:
```diff
@@ -0,0 +1,95 @@
+        print(f".env file not found at: {path}")
+        sys.exit(1)
+
+    with path.open(encoding="utf-8") as f:
+        for raw in f:
+            line = raw.strip()
+            if not line or line.startswith("#"):
+                continue
+            if "=" not in line:
+                # skip malformed lines quietly (matches typical .env loaders)
+                continue
+            key, value = line.split("=", 1)
+            key = key.strip()
+            value = value.strip()
+
+            # remove surrounding single/double quotes if present
+            if (value.startswith('"') and value.endswith('"')) or (
+                value.startswith("'") and value.endswith("'")
+            ):
+                value = value[1:-1]
+
+            os.environ[key] = value
```
even though this is probably a correct manual adding of the env variables, we already use dotenv package to load env vars in our project so you can instead do:
```python
from dotenv import load_dotenv

# Load environment variables
load_dotenv(override=True)
```

omar-abdelgawad reviewed (commented):
Great Job! I only have some comments that we can merge after resolving:
1. please delete the two sh and cmd files that are replaced by this script.
2. please add to our Makefile a command for running this script called `setup_and_pull` which we can run by `make setup_and_pull`. you can do that using:
```console
make setup_and_pull
```
and the makefile should contain somthing like this:
```Makefile
setup_and_pull:
	uv run setup_and_pull_general.py
```
3. address the comment in the file about using `load_dotenv` for loading env vars.

Abdelaziz-Serour committed:
Update setup_and_pull_general.py

use dotenv

Abdelaziz-Serour committed:
Update README.md

The new command to run the setup_and_pull_general.py file

Abdelaziz-Serour committed:
Delete setup_and_pull.cmd

Abdelaziz-Serour committed:
Delete setup_and_pull.sh

Abdelaziz-Serour committed:
Update Makefile

Added a new setup_and_pull Makefile target to simplify running the environment setup and model pulling script

omar-abdelgawad committed:
Formatted Script

omar-abdelgawad reviewed (approved):
LGTM 
![ship-it-cat-gif](https://i.fluffy.cc/Xr1jnB3r6dWrfwqPwNdrpnF9zW1lVp0F.gif)
